packages/forms-web-app/src/controllers/examination/submitting-for/utils/get-redirect-url.js

```diff
@@ -0,0 +1,26 @@
+const { isQueryModeEdit } = require('../../../utils/is-query-mode-edit');
+const {
+	routesConfig: {
+		examination: {
+			directory,
+			pages: { checkYourAnswers, nameAgent, nameMyself, nameOrganisation, submittingFor }
+		}
+	}
+} = require('../../../../routes/config');
+
+const getRedirectUrl = (query, selectedSubmittingForValue) => {
+	if (isQueryModeEdit(query)) return `${directory}${checkYourAnswers.route}`;
+
+	switch (selectedSubmittingForValue) {
+		case submittingFor.options[1].value:
+			return `${directory}${nameMyself.route}`;
+		case submittingFor.options[2].value:
+			return `${directory}${nameOrganisation.route}`;
+		case submittingFor.options[3].value:
+			return `${directory}${nameAgent.route}`;
+		default:
```

Comment:
Throw an exception, since this is an exceptional case.

And throw it outside the switch, it would be more readable. Otherwise it sounds like "the default case is exceptional".


https://stackoverflow.com/questions/2567932/throwing-exception-vs-returning-null-value-with-switch-statement

Reply:
Happy to keep it but want to ask the question.